Unit test fixes to get tests to pass under mono

mcraveiro opened the pull request:
- replace backslashes with Path.Combine where possible;
  where not possible (attributes) use slash.
- ensure case of files and directories matches file system.
- handle rooted paths in unix.
- do not use SelectSingleNode when removing due to undefined
  behaviour[1].

[1] http://stackoverflow.com/questions/2496614/trying-to-use-removechild-on-xmlnodelist-messes-up-my-xmlnode-collection

avao commented:
I have merged develop into master and deleted develop. Please resubmit pull request

mcraveiro committed:
Unit test fixes to get tests to compile under mono

- replace backslashes with Path.Combine where possible;
  where not possible (attributes) use slash.
- ensure case of files and directories matches file system.
- handle rooted paths in unix.
- do not use SelectSingleNode when removing due to undefined
  behaviour[1].

[1] http://stackoverflow.com/questions/2496614/trying-to-use-removechild-on-xmlnodelist-messes-up-my-xmlnode-collection
